src/Mafin.Wizard.Api/ResultFileFactory.cs

```diff
@@ -0,0 +1,65 @@
+using System.Text.RegularExpressions;
+using Mafin.Wizard.Api.Constants;
+using Mafin.Wizard.Api.Extensions;
+using Mafin.Wizard.Api.Models;
+using Scriban;
+
+namespace Mafin.Wizard.Api;
+
+internal class ResultFileFactory
+{
+    private const string PathPattern = "#### path: (.*?) ####";
+    private const string FileNamePattern = "#### name: (.*?) ####";
+
+    public FileInfoRecord CreateResultFileRecord(ScribanTemplate template, TemplateContext context)
+    {
+        var renderedTemplate = RenderTemplate(template, context);
+        var filename = GetFileName(template.Name, ref renderedTemplate);
+        var directory = GetDirectory(ref renderedTemplate);
+        foreach (var extension in ScribanFileExtensions.ScriptExtensions.Where(extension => filename.EndsWith($".{extension}", StringComparison.OrdinalIgnoreCase)))
+        {
+            filename = filename.TrimEnd($".{extension}");
+            break;
```

Comment:
It looks like you're breaking the loop after finding the first result. Should it be just FirstOrDefault() intead then?
Also, in this case the caller method looks fishy, as it puts that single result into IEnumerable and then converts to List.
Could you, please, re-check the purpose here?

To me it looks like you don't need more than a single result, but maybe I'm missing something.

Reply:
fixed. thanks